src/core/DataFile.cpp

```diff
@@ -901,6 +901,29 @@ void DataFile::upgrade_1_1_91()
 }
 
 
+void DataFile::upgrade_1_2_0_rc3()
+{
+	// Upgrad from earlier bbtrack beat note behaviour of adding
+	// steps if a note is placed after the last step.
+	QDomNodeList list = elementsByTagName( "bbtrack" );
+	for( int i = 0; !list.item( i ).isNull(); ++i )
+	{
+		list = elementsByTagName( "pattern" );
+		for( int i = 0; !list.item( i ).isNull(); ++i )
+		{
+			int patternLength, steps;
+			QDomElement el = list.item( i ).toElement();
+			for( int i = 0; !list.item( i ).isNull(); ++i )
+			{
+				patternLength = el.attribute( "len" ).toInt();
```

Comment:
OK. I have no more tests do do here but as it changes the behaviour of a central tool maybe leave it up for a while before merge. Is it too optimistic to think that someone is going to bother to take it for a spin? We have quite a list of pull requests to go through. :)